test/providers/provider_level_zero.cpp

```diff
@@ -326,21 +326,66 @@ struct umfLevelZeroProviderTest : umf_test::test {
         ASSERT_EQ(ze_result, ZE_RESULT_SUCCESS);
     }
 
-    ze_driver_handle_t hDriver;
-    ze_device_handle_t hDevice;
-    ze_context_handle_t hContext;
-    void *zeDlHandle;
+  private:
+    ze_device_handle_t hDevice_;
+    ze_context_handle_t hContext_;
+};
+
+using LevelZeroProviderTestParams =
+    std::tuple<level_zero_memory_provider_params_t, MemoryAccessor *>;
+
+struct umfLevelZeroProviderTest
+    : umf_test::test,
+      ::testing::WithParamInterface<LevelZeroProviderTestParams> {
+
+    void SetUp() override {
+        test::SetUp();
+
+        auto [l0_params, accessor] = this->GetParam();
+        params = l0_params;
+        hDevice = (ze_device_handle_t)params.level_zero_device_handle;
+        hContext = (ze_context_handle_t)params.level_zero_context_handle;
+
+        if (nullptr == hDevice) {
+            GTEST_SKIP() << "Test skipped, no GPU devices found";
```

Comment:
I think we may fail the test actually - we have a specific CMake flag to enable these tests (`UMF_BUILD_GPU_TESTS`, OFF by default). I believe we prefer not to have a "silent skips" (gtest doesn't say a word about skips).

Reply:
Done